crates/sui-move/src/build.rs

```diff
@@ -41,7 +46,14 @@ impl Build {
         build_config: MoveBuildConfig,
     ) -> anyhow::Result<()> {
         let rerooted_path = base::reroot_path(path.clone())?;
-        let build_config = resolve_lock_file_path(build_config, path)?;
+        let build_config = resolve_lock_file_path(build_config, path.clone())?;
+        let lookup = resolve_manifest_lookup(path)?;
+        let _published_at = if let Some(_addr) = lookup.get(PUBLISHED_AT_MANIFEST_FIELD) {
+            // TODO parse
+        } else {
+            // TODO decide how to handle / fail when there is no `published-at` field.
+        };
```

Comment:
In https://github.com/MystenLabs/sui/issues/8343 we mention running the logic on `client publish`. I noticed that if I don't include the manifest parsing in `move build`, we'll see:

> Warning: unknown field name found. Expected one of [name, version, authors, license], but found 'published-at'

Thoughts on the role of `publish-at` at `move build` time? Not sure if we can suppress this, so we could just parse and ignore during build once we do real things with the value (for `client publish`).

Reply:
My vote is for parse and ignore here!